include/evmc/evmc.h

```diff
@@ -190,18 +190,23 @@ struct evmc_message
 };
 
 
-/** The transaction and block data for execution. */
+/// The additional information about the transaction being executed.
 struct evmc_tx_context
 {
-    evmc_uint256be tx_gas_price;     /**< The transaction gas price. */
-    evmc_address tx_origin;          /**< The transaction origin account. */
-    evmc_address block_coinbase;     /**< The miner of the block. */
-    int64_t block_number;            /**< The block number. */
-    int64_t block_timestamp;         /**< The block timestamp. */
-    int64_t block_gas_limit;         /**< The block gas limit. */
-    evmc_uint256be block_difficulty; /**< The block difficulty. */
-    evmc_uint256be chain_id;         /**< The blockchain's ChainID. */
-    evmc_uint256be block_base_fee;   /**< The block base fee per gas (EIP-1559, EIP-3198). */
+    evmc_address origin;       ///< The transaction origin account.
+    evmc_uint256be gas_price;  ///< The transaction gas price.
+};
+
+/// The additional information about the block in which execution happens.
+struct evmc_block_context
+{
+    int64_t number;             ///< The block number.
+    int64_t timestamp;          ///< The block timestamp.
+    int64_t gas_limit;          ///< The block gas limit.
+    evmc_uint256be chain_id;    ///< The blockchain's ChainID.
+    evmc_uint256be base_fee;    ///< The block base fee per gas (EIP-1559, EIP-3198).
+    evmc_uint256be difficulty;  ///< The block difficulty.
```

Comment:
I'd rename `difficulty` to `random` for compatibility with the upcoming [EIP-4399](https://eips.ethereum.org/EIPS/eip-4399).